x/ecocredit/server/basket/util.go

```diff
@@ -0,0 +1,59 @@
+package basket
+
+import (
+	"context"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	"github.com/regen-network/regen-ledger/types/math"
+)
+
+func (k Keeper) GetBasketBalanceMap(ctx context.Context) (map[uint64]math.Dec, error) {
+	itr, err := k.stateStore.BasketTable().List(ctx, api.BasketPrimaryKey{})
+	if err != nil {
+		return nil, err
+	}
+	defer itr.Close()
+
+	batchDenomToId := make(map[string]uint64)     // map of a batch denom to batch id
+	batchIdToBalance := make(map[uint64]math.Dec) // map of a basket batch_id to balance
+	for itr.Next() {
+		basket, err := itr.Value()
+		if err != nil {
+			return nil, err
+		}
+
+		bb, err := k.stateStore.BasketBalanceTable().Get(ctx, basket.Id, basket.BasketDenom)
+		if err != nil {
+			return nil, err
+		}
+
+		amount, err := math.NewDecFromString(bb.Balance)
+		if err != nil {
+			return nil, err
+		}
+
+		var batchID uint64
+		if _, ok := batchDenomToId[basket.BasketDenom]; !ok {
+			bInfo, err := k.coreStore.BatchInfoTable().GetByBatchDenom(ctx, basket.BasketDenom)
```

Comment:
Same here. A batch should never exist with a basket denom as the batch denom. I think you want to loop through the batches within a basket and get the credit balance of each batch to check against the total supply. We don't need to check the balance of the basket denom against anything in core but we do need to check the balance of the credits within baskets against the total supply.

Reply:
> Same here. A batch should never exist with a basket denom as the batch denom. I think you want to loop through the batches within a basket and get the credit balance of each batch to check against the total supply. We don't need to check the balance of the basket denom against anything in core but we do need to check the balance of the credits within baskets against the total supply.

updated